src/hotspot/share/services/diagnosticArgument.cpp

```diff
@@ -183,20 +183,17 @@ template <> void DCmdArgument<char*>::parse_value(const char* str,
   if (str == nullptr) {
```

Comment:
Do we need to call free here? Not sure if str is actually ever null in practice.

Reply:
I have the same question.  Why don't we call free if _value is non-null?

Reply:
Not sure str can ever actually be null here, but if it is then yes we should also free. Thanks for catching that.

Reply:
Turns out it is trivially easy for it to be null - just pass a settable option that doesn't have "=x". Test has been updated to cover missing cases.